### ISSUE TITLE
Single object file for multiple targets

### ISSUE THREAD

**Issue by g3cko** [ispc/ispc#1850]
Am I overlooking anything that would allow a single object file if I specify multiple `--target=` options?    Our current build system doesn't have great support for 1 - N.

**Comment by pbrubaker** [comment]
No, you're not overlooking anything.  I ran into a situation where this
would have been useful recently as well, and mentioned it internally.  I'll
create an issue so it can be looked at.

**Comment by g3cko** [comment]
Thanks for the speedy reply!  Ideally I would be able to work around this more easily from our build system but for the alternative, it sounds like this could be useful for `ispc` to have too.

**Comment by dbabokin** [comment]
Initially it was caused by limitations in LLVM, but now LLVM allows targeting different CPUs in a single compilation module. Though, it will require non-trivial changes on ISPC side at this point. But I think we need to implement it.
Thanks for bringing this up!

**Comment by clusty** [comment]
This is indeed very annoying to integrate inside a build system. I managed to get around it by glueing the different object files together using `ld -r -o`

**Comment by pbrubaker** [comment]
This won't work on MSVC sadly.  It's only possible to combine multiple object files into a static library.

**Comment by clusty** [comment]
> This won't work with the MSVC linker sadly. It's only possible to combine multiple object files into a static library. (Which is what ld is doing really...)
> 
> [@dbabokin](https://github.com/dbabokin) could we do this as a post process once all the targets are generated?
 
You mean the ld trick or is it a fundamental platform limitation with merging object files ?

**Comment by dbabokin** [comment]
Ideally we should do that within single compilation unit - i.e. compile functions for different targets within single LLVM bytecode module. Post-processing is also an option.
